frame/staking/src/pallet/mod.rs

```diff
@@ -932,26 +933,34 @@ pub mod pallet {
 		/// the funds out of management ready for transfer.
 		///
 		/// No more than a limited number of unlocking chunks (see `MaxUnlockingChunks`)
-		/// can co-exists at the same time. In that case, [`Call::withdraw_unbonded`] need
-		/// to be called first to remove some of the chunks (if possible).
+		/// can co-exists at the same time. If there are no unlocking chunks slots available
+		/// [`Call::withdraw_unbonded`] is called to remove some of the chunks (if possible).
 		///
 		/// If a user encounters the `InsufficientBond` error when calling this extrinsic,
 		/// they should call `chill` first in order to free up their bonded funds.
 		///
 		/// Emits `Unbonded`.
 		///
 		/// See also [`Call::withdraw_unbonded`].
-		#[pallet::weight(T::WeightInfo::unbond())]
+		#[pallet::weight(
+            T::WeightInfo::withdraw_unbonded_kill(SLASHING_SPANS_AUTO_WITHDRAW) + T::WeightInfo::unbond())
+        ]
 		pub fn unbond(
 			origin: OriginFor<T>,
 			#[pallet::compact] value: BalanceOf<T>,
 		) -> DispatchResult {
-			let controller = ensure_signed(origin)?;
+			let controller = ensure_signed(origin.clone())?;
+
+
+			// ensure that there's chunk slots available by requesting the staking interface to
+			// withdraw chunks older than `BondingDuration`, if there are no more unlocking chunks
+			// slots available.
+			if Self::chunk_slots_filled(&controller)? == T::MaxUnlockingChunks::get() as usize {
+				Self::withdraw_unbonded(origin, SLASHING_SPANS_AUTO_WITHDRAW)
```

Comment:
I'm still unsure what's the `num_slashing_spans` that should be passed here. If the nominator has been slashed and we pass `SLASHING_SPANS_AUTO_WITHDRAW = 0` and the active stake is lower than the existential deposit, the call to `fn kill_stash` will fail. We query `Self::SlashingSpans` for the account but it seems that it is not possible to pass values to `T::WeightInfo::withdraw_unbonded_kill()` when the vars are in the function scope.